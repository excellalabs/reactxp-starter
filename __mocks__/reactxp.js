
import * as React from 'react'

const identity = () => jest.fn((i) => i)
const asyncIdent = (i) => jest.fn(() => Promise.resolve(i))
const noop = jest.fn()

const subscribableEvent = () => {
  return {
    dispose: jest.fn(),
    subscribe: jest.fn(() => { unsubscribe: jest.fn() }),
    unsubscribe: jest.fn(),
    fire: jest.fn()
  }
}

// Components
export const ActivityIndicator = 'RX.ActivityIndicator'
export const Button = 'RX.Button'
export const Picker = 'RX.Picker'
export const GestureView = 'RX.GestureView'
export const Link = 'RX.Link'
export const Text = 'RX.Text'
export const TextInput = 'RX.TextInput'
export const View = 'RX.View'

// Objects
export const Alert = {
  show: jest.fn()
}

export const UserInterface = {
  setMainView: jest.fn(),
  useCustomScrollbars: jest.fn(),
  isHighPixelDensityScreen: jest.fn(() => false),
  getPixelRatio: jest.fn(() => 1)
}

export const Modal = {
  isDisplayed: jest.fn(() => false),
  show: jest.fn(),
  dismiss: jest.fn(),
  dismissAll: jest.fn()
}

export const Popup = {
  show: jest.fn(() => false),
  autoDismiss: jest.fn(),
  dismiss: jest.fn(),
  dismissAll: jest.fn()
}

export const Linking = {
  getInitialUrl: asyncIdent(),
  deepLinkRequestEvent: subscribableEvent(),
  openUrl: asyncIdent(),
  launchSms: asyncIdent(),
  launchEmail: asyncIdent()
}

export const Accessibility = {
  isScreenReaderEnabled: jest.fn(() => false),
  announceForAccessibility: jest.fn(),
  screenReaderChangedEvent: subscribableEvent()
}

export const Clipboard = {
  setText: jest.fn(),
  getText: asyncIdent('text')
}

export const Storage = {
  getItem: asyncIdent(),
  setItem: asyncIdent(),
  removeItem: asyncIdent(),
  clear: asyncIdent()
}

export const Location = {
  isAvailable: jest.fn(() => false),
  setConfiguration: jest.fn(),
  getCurrentPosition: asyncIdent({ 
    coords: {
      accuracy: 0,
      altitude: null,
      altitudeAccuracy: null,
      heading: null,
      latitude: 0,
      longitude: 0,
      speed: null
    },
    timestamp: 0
  }),
  watchPosition: asyncIdent(0),
  clearWatch: jest.fn()
}

export const Network = {
  isConnected: asyncIdent(false),
  getType: asyncIdent(0),
  connectivityChangedEvent: subscribableEvent()
}

export const Platform = {
  getType: () => 'test'
}

export const Input = {
  backButtonEvent: subscribableEvent(),
  keyDownEvent: subscribableEvent(),
  keyUpEvent: subscribableEvent(),
}

export const StatusBar = {
  isOverlay: jest.fn(() => false),
  setHidden: jest.fn(),
  setBarStyle: jest.fn(),
  setNetworkActivityIndicatorVisible: jest.fn(),
  setBackgroundColor: jest.fn(),
  setTranslucent: jest.fn(),
}

export const Styles = {
  combine: jest.fn((...args) => args),
  createViewStyle: identity(),
  createAnimatedViewStyle: identity(),
  createScrollViewStyle: identity(),
  createButtonStyle: identity(),
  createWebViewStyle: identity(),
  createTextStyle: identity(),
  createAnimatedTextStyle: identity(),
  createTextInputStyle: identity(),
  createAnimatedTextInputStyle: identity(),
  createImageStyle: identity(),
  createAnimatedImageStyle: identity(),
  createLinkStyle: identity(),
  createPickerStyle: identity(),
  getCssPropertyAliasesCssStyle: {},
}

export const UserPresence = {
  isUserPresent: jest.fn(() => false),
  userPresenceChangedEvent: subscribableEvent()
}

export const Animated = {
  Image: 'RX.Animated.Image',
  Text: 'RX.Animated.Text',
  TextInput: 'RX.Animated.TextInput',
  View: 'RX.Animated.View',
}

export const International = {
  allowRTL: jest.fn(),
  forceRTL: jest.fn(),
  isRTL: jest.fn(() => true),
}

export const Component = React.Component