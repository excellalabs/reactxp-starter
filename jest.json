{
  "preset": "react-native",
  "transform": {
    "\\.(ts|tsx)$": "ts-jest/preprocessor.js",
    "\\.(js|jsx)$": "babel-jest"
  },
  "testRegex": "(src|storybook)/.*(/__tests__/.*|\\.(spec))\\.(ts|tsx|js)$",
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js",
    "json"
  ],
  "globals": {
    "ts-jest": {
      "tsConfigFile": "tsconfig.json"
    }
  },
  "moduleNameMapper": {
    "^commonStyles$": "<rootDir>/src/commonStyles/index.ts",
    "^commonStyles/(.*)$": "<rootDir>/src/commonStyles/$1.ts",
    "^components/(.*)$": "<rootDir>/src/components/$1/index.ts",
    "^containers/(.*)$": "<rootDir>/src/containers/$1/index.ts",
    "^helpers/(.*)$": "<rootDir>/src/helpers/$1.ts",
    "^queries/(.*)$": "<rootDir>/src/queries/$1.ts",
    "^redux-store$": "<rootDir>/src/redux-store/index.ts",
    "^actions/(.*)$": "<rootDir>/src/redux-store/actions/$1.ts",
    "^thunks/(.*)$": "<rootDir>/src/redux-store/thunks/$1.ts",
    "^routes/(.*)$": "<rootDir>/src/routes/$1/index.ts",
    "^services/(.*)$": "<rootDir>/src/services/$1.ts"
  },
  "collectCoverage": true,
  "collectCoverageFrom": [
    "src/**/*.{ts,tsx}",
    "!src/index.tsx",
    "!src/**/*.d.ts",
    "!src/redux-store/index.ts",
    "!src/redux-store/rootReducer.ts",
    "!src/redux-store/store.ts",
    "!src/routes/**/*"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 20,
      "functions": 20,
      "lines": 20,
      "statements": 20
    }
  }
}