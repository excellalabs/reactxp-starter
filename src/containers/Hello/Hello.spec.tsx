
import 'jest'
import 'react-native'
import * as React from 'react'
import * as RX from 'reactxp'
import { shallow, configure } from 'enzyme'

import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

import OriginalHello, { HelloBase as OriginalHelloBase } from './Hello'

// Note: test renderer must be required after react-native.
import * as renderer from 'react-test-renderer'

const importModule = (): typeof OriginalHello => {
  return require('.').default
}
const importBase = (): typeof OriginalHelloBase => {
  return require('./Hello').HelloBase
}

const mockedComponents = [
]

const defaultMocks = [
]

const allMocks = mockedComponents.concat(defaultMocks)

describe('<Hello />', () => {
  beforeEach(() => {
    jest.resetModules()
    mockedComponents.forEach(component => {
      jest.setMock(component, component)
    })
    defaultMocks.forEach(component => {
      jest.setMock(component, jest.genMockFromModule(component))
    })
  })

  afterEach(() => {
    allMocks.forEach(component => {
      jest.unmock(component)
    })
  })

  describe('base', () => {
    const names = [
      'World',
      'React',
      'Kevin',
      'Doguhan',
      'Stuart'
    ]

    describe('snapshots', () => {
      names.forEach(name => {
        it(`renders with the name ${name}`, () => {
          const HelloBase = importBase()
          const tree = renderer.create(
            <HelloBase name={name} chooseRandomName={jest.fn()} />
          )

          expect(tree).toMatchSnapshot()
        })
      })
    })

    // describe('user clicks the "Random Name" button', () => {
    //   it('should call chooseRandomName', () => {
    //     const callback = jest.fn()
    //
    //     const HelloBase = importBase()
    //     const tree = shallow(
    //       <HelloBase name={'World'} chooseRandomName={callback} />
    //     )
    //
    //     expect(callback).toHaveBeenCalledTimes(0)
    //
    //     tree.find(RX.Button).simulate('click')
    //     expect(callback).toHaveBeenCalledTimes(1)
    //   })
    // })
  })
})
