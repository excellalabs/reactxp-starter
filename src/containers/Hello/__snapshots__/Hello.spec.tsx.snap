// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hello /> base snapshots renders with the name Doguhan 1`] = `
<RX.View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RX.Text
    style={
      Object {
        "color": "#110022",
        "fontSize": 18,
        "paddingBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Hello, 
    Doguhan
  </RX.Text>
  <RX.Button
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#330066",
          "borderRadius": 5,
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        },
      ]
    }
  >
    <RX.Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Random Name
    </RX.Text>
  </RX.Button>
</RX.View>
`;

exports[`<Hello /> base snapshots renders with the name Kevin 1`] = `
<RX.View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RX.Text
    style={
      Object {
        "color": "#110022",
        "fontSize": 18,
        "paddingBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Hello, 
    Kevin
  </RX.Text>
  <RX.Button
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#330066",
          "borderRadius": 5,
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        },
      ]
    }
  >
    <RX.Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Random Name
    </RX.Text>
  </RX.Button>
</RX.View>
`;

exports[`<Hello /> base snapshots renders with the name React 1`] = `
<RX.View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RX.Text
    style={
      Object {
        "color": "#110022",
        "fontSize": 18,
        "paddingBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Hello, 
    React
  </RX.Text>
  <RX.Button
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#330066",
          "borderRadius": 5,
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        },
      ]
    }
  >
    <RX.Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Random Name
    </RX.Text>
  </RX.Button>
</RX.View>
`;

exports[`<Hello /> base snapshots renders with the name Stuart 1`] = `
<RX.View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RX.Text
    style={
      Object {
        "color": "#110022",
        "fontSize": 18,
        "paddingBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Hello, 
    Stuart
  </RX.Text>
  <RX.Button
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#330066",
          "borderRadius": 5,
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        },
      ]
    }
  >
    <RX.Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Random Name
    </RX.Text>
  </RX.Button>
</RX.View>
`;

exports[`<Hello /> base snapshots renders with the name World 1`] = `
<RX.View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RX.Text
    style={
      Object {
        "color": "#110022",
        "fontSize": 18,
        "paddingBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Hello, 
    World
  </RX.Text>
  <RX.Button
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#330066",
          "borderRadius": 5,
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        },
      ]
    }
  >
    <RX.Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Random Name
    </RX.Text>
  </RX.Button>
</RX.View>
`;
